evalml/objectives/standard_metrics.py

```diff
@@ -901,6 +902,40 @@ def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
         )
 
 
+class SilhouetteCoefficient(UnsupervisedLearningObjective):
+    """Silhouette Coefficient for clustering. Does not require ground truth values."""
+
+    name = "Silhouette Coefficient"
+    greater_is_better = True
+    score_needs_proba = False
+    perfect_score = 1.0
+    is_bounded_like_percentage = False
+    expected_range = [-1, 1]
+    requires_y_true = False
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
+        """Objective function for silhouette coefficient for clustering."""
+        try:
+            return metrics.silhouette_score(X, y_predicted, metric="euclidean")
+        except ValueError:
+            return 0
+
+
+class AdjustedRandScore(UnsupervisedLearningObjective):
+    """Rand Score for clustering, adjusted for chance. Does require ground truth values."""
+
+    name = "Adjusted Rand Score"
+    greater_is_better = True
+    score_needs_proba = False
+    perfect_score = 1.0
+    is_bounded_like_percentage = True
+    expected_range = [0, 1]
+
+    def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
```

Comment:
Where do the ground truth labels come from? At this point is this a classification objective?

Reply:
While less common, there is a chance a user may have ground-truth information about their data. Alternatively, having a metric that takes true labels will be useful for our performance tests in the future.

The distinction between this metric and a standard classification metric is that this metric is permutation-independent. You can see an example in the tests, where the cluster numbers in `y_true` and `y_predicted` are swapped, but the score is 1.0 regardless.